ui/core.js

```diff
@@ -126,6 +128,67 @@ $.fn.extend( {
 				$( this ).removeAttr( "id" );
 			}
 		} );
+	},
+
+	form: function() {
+		var form;
+
+		if ( this[ 0 ].form && typeof this[ 0 ].form !== "string" ) {
+			return this.pushStack( $( this[ 0 ].form ) );
+		} else if ( !this[ 0 ].form ) {
+			return this.pushStack( [] );
+		}
+
+		// Support: IE8 only ( the rest of the method )
+		// IE8 supports the form property, but not the form attribute, worse yet if you supply the
+		// form attribute it overwrites the form property with the string. Other supported browsers
+		// like all other IE's and Android 2.3 support the form attribute not at all or partially.
+		// We don't care in those cases because they still always return an element. We are not
+		// trying to fix the form attribute here, only deal with the prop supplying a string.
+		form = this.closest( "form" );
+		if ( form.length ) {
+			return this.pushStack( form );
+		}
+		return this.pushStack( [] );
+	},
+
+	labels: function() {
+		var ancestor, selector, id, labels, ancestors;
+
+			// Check control.labels first
+			if ( this[ 0 ].labels !== undefined && this[ 0 ].labels.length > 0 ) {
+				return this.pushStack( this[ 0 ].labels );
+			}
+
+			// Support: IE <= 11, FF <= 37, Android <=2.3 only
+			// Above browsers do not support control.labels everything below is to support them
+			// as well as document fragments control.labels does not work on document fragments anywhere
+			labels = this.parents( "label" );
+
+			// Look for the label based on the id
+			id = this.attr( "id" );
+			if ( id ) {
+
+				// We don't search against the document in case the element
+				// is disconnected from the DOM
+				ancestor = this.parents().last();
+
+				// get a full set of top level ancestors
+				ancestors = ancestor.add( ancestor.length ? ancestor.siblings() : this.siblings() );
+
+				// Create a selector for the label based on the id
+				selector = "label[for='" + this.attr( "id" ).replace( $.ui.escapeId, "\\$1" ) + "']";
+
+				// Support: Core 1.7
+				// Once we drop support for 1.7 we can just use .addBack( selector ) here
+				// Check both the ancestors and the contents of the ancestors for matching labels
+				labels = labels.add( ancestors.filter( selector ) );
+				labels = labels.add( ancestors.find( selector ) );
+
+			}
+
+			// Return whatever we have found for labels
+			return this.pushStack( $.unique( labels ) );
```

Comment:
In what situations are the elements not sorted properly? This needs a support comment.